mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -37,6 +38,39 @@ def _format_variable_string(variables: Dict[str, Any], eval_df, indx) -> str:
     )
 
 
+# Function to extract Score and Justification
+def _extract_score_and_justification(output):
+    if (
+        isinstance(output, dict)
+        and "candidates" in output
+        and isinstance(output["candidates"], list)
+        and output["candidates"]
+    ):
+        text = output["candidates"][0]["text"]
+    else:
+        text = output.candidates[0].text
```

Comment:
can we add a comment about why we need to have this logic? 

also do we need to have this logic? if this is a discrepancy between gateway and calling openai directly, we change our implementation of calling openai directly.